docs/sources/flow/reference/components/remote.vault.md

```diff
@@ -0,0 +1,319 @@
+auth.token | [auth.token][] | Authenticate to Vault with a token. | no
+auth.approle | [auth.approle][] | Authenticate to Vault using AppRole. | no
+auth.aws | [auth.aws][] | Authenticate to Vault using AWS. | no
+auth.azure | [auth.azure][] | Authenticate to Vault using Azure. | no
+auth.gcp | [auth.gcp][] | Authenticate to Vault using GCP. | no
+auth.kubernetes | [auth.kubernetes][] | Authenticate to Vault using Kubernetes. | no
+auth.ldap | [auth.ldap][] | Authenticate to Vault using LDAP. | no
+auth.userpass | [auth.userpass][] | Authenticate to Vault using a username and password. | no
+auth.custom | [auth.custom][] | Authenticate to Vault with custom authentication. | no
+
+Exactly one `auth.*` block **must** be provided, otherwise the component will
+fail to load.
+
+[client_options]: #client_options-block
+[auth.token]: #authtoken-block
+[auth.approle]: #authapprole-block
+[auth.aws]: #authaws-block
+[auth.azure]: #authazure-block
+[auth.gcp]: #authgcp-block
+[auth.kubernetes]: #authkubernetes-block
+[auth.ldap]: #authldap-block
+[auth.userpass]: #authuserpass-block
+[auth.custom]: #authcustom-block
+
+### client_options block
+
+The `client_options` block customizes the connection to vault.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`min_retry_wait` | `duration` | Minimum time to wait before retrying failed requests. | `"1000ms"` | no
+`max_retry_wait` | `duration` | Maximum time to wait before retrying failed requests. | `"1500ms"` | no
+`max_retries` | `int` | Maximum number of times to retry after a 5xx error. | `2` | no
+`timeout` | `duration` | Maximum time to wait before a request times out. | `"60s"` | no
+
+Requests which fail due to server errors (HTTP 5xx error codes) can be retried.
+The `max_retries` argument specifies how many times to retry failed requests.
+The `min_retry_wait` and `max_retry_wait` arguments specify how long to wait
+before retrying. The wait period starts at `min_retry_wait` and exponentially
+increases up to `max_retry_wait`.
+
+Other types of failed requests, including HTTP 4xx error codes, are not
+retried.
+
+If the `max_retries` argument is set to `0`, failed requests are not retried.
+
+### auth.token block
+
+The `auth.token` block authenticates each request to Vault using a
+token.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`token` | `secret` | Authentication token to use. | | yes
+
+### auth.approle block
+
+The `auth.token` block auhenticates to Vault using the [AppRole auth
+method][AppRole].
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`role_id` | `string` | Role ID to authenticate as. | | yes
+`secret` | `secret` | Secret to authenticate with. | | yes
+`wrapping_token` | `bool` | Whether to [unwrap][] the token. | `false` | no
+`mount_path` | `string` | Mount path for the login. | `"approle"` | no
```

Comment:
I think we're not setting the `mount_path` default in any of the authentication methods here.